src/pybind/mgr/dashboard/controllers/mgr_modules.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import
+
+from . import ApiController, RESTController, Endpoint, UpdatePermission
+from .. import mgr
+from ..security import Scope
+from ..services.ceph_service import CephService
+from ..services.exception import handle_send_command_error
+from ..tools import str_to_bool
+
+
+@ApiController('/mgr/module', Scope.CONFIG_OPT)
+class MgrModules(RESTController):
+    managed_modules = ['telemetry']
+
+    @handle_send_command_error('mgr_modules')
+    def list(self):
+        result = []
+        modules = CephService.send_command('mon', 'mgr module ls')
+        for name in modules['enabled_modules']:
+            if name in self.managed_modules:
+                result.append({'name': name, 'enabled': True})
+        for module_info in modules['disabled_modules']:
+            if module_info['name'] in self.managed_modules:
+                result.append({'name': module_info['name'], 'enabled': False})
+        return result
+
+
+class MgrModuleRESTController(RESTController):
+    module_name = None
+
+    def list(self):
+        """
+        Retrieve the values of the persistent configuration settings.
+        :return: The values of the module options.
+        :rtype: dict
+        """
+        options = self._get_module_options()
+        result = {}
+        for name, option in options.items():
+            result[name] = mgr.get_module_option_ex(self.module_name, name,
+                                                    option['default_value'])
+        return result
+
+    def bulk_set(self, config):
+        """
+        Set the values of the persistent configuration settings.
+        :param config: The values of the module options to be stored.
+        :type config: dict
+        """
+        options = self._get_module_options()
+        for name, option in options.items():
+            if name in config:
+                mgr.set_module_option_ex(self.module_name, name, config[name])
+
+    @Endpoint('POST', json_response=False)
+    @UpdatePermission
+    @handle_send_command_error('mgr_modules')
+    def enable(self):
+        """
+        Enable the Ceph Mgr module.
+        """
+        CephService.send_command(
+            'mon', 'mgr module enable', module=self.module_name)
+
+    @Endpoint('POST', json_response=False)
+    @UpdatePermission
+    @handle_send_command_error('mgr_modules')
+    def disable(self):
+        """
+        Disable the Ceph Mgr module.
+        """
+        CephService.send_command(
+            'mon', 'mgr module disable', module=self.module_name)
+
+    @handle_send_command_error('mgr_modules')
+    def _get_module_info(self):
+        """
+        Helper function to get detailed module information.
+        :return: The module information, e.g. module name, can run,
+            error string and available module options.
+        :rtype: dict or None
+        """
+        dump = CephService.send_command(
+            'mon', 'mgr dump', module=self.module_name)
```

Comment:
You can get the MgrMap from the in-memory cache in Mgr (`mgr.get('mgr_map')`). In fact, we should only use `mgr.send_command()` for commands that cannot be obtained from Mgr cached info, and that modify the cluster.

Reply:
Fixed.